modules/gateway.go

```diff
@@ -162,9 +162,15 @@ type (
 		// Connect establishes a persistent connection to a peer.
 		Connect(NetAddress) error
 
+		// ConnectManual is a Connect wrapper for a user-initiated Connect
+		ConnectManual(NetAddress) error
+
 		// Disconnect terminates a connection to a peer.
 		Disconnect(NetAddress) error
 
+		// DisconnectManual is a Disconnect wrapper for a user-initiated disconnect
+		DisconnectManual(NetAddress) error
```

Comment:
why do we need these? Wouldn't a `Blacklist` method make more sense? It's not clear to me that a user-initiated disconnect always implies that the user wants to blacklist the peer.

As always, we may want to check how bitcoind solves these problems.

Reply:
I think bitcoind has a [`setban`](https://bitcoin.stackexchange.com/questions/43236/manually-banning-nodes-through-cli) function that allows a user to ban a node for a certain period of time.
If we don't like this behaviour we can also change this PR to introduce a new endpoint. I started it because it was one of the todos.

Reply:
@DavidVorick any thoughts on that?